nano/core_test/block_store.cpp

```diff
@@ -1012,30 +1012,30 @@ TEST (mdb_block_store, sideband_height)
 	nano::block_builder builder;
 	auto transaction (store.tx_begin_write ());
 	store.initialize (transaction, ledger.cache, nano::dev::constants);
-	nano::work_pool pool{ nano::dev::network_params.network, std::numeric_limits<unsigned>::max () };
+	nano::test::start_stop_container<nano::work_pool> pool{ nano::dev::network_params.network, std::numeric_limits<unsigned>::max () };
 	auto send = builder
 				.send ()
 				.previous (nano::dev::genesis->hash ())
 				.destination (nano::dev::genesis_key.pub)
 				.balance (nano::dev::constants.genesis_amount - nano::Gxrb_ratio)
 				.sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
-				.work (*pool.generate (nano::dev::genesis->hash ()))
+				.work (*pool.obj.generate (nano::dev::genesis->hash ()))
```

Comment:
Why do we need this strange pattern of `pool.obj...`? Keep it as just `pool`. Simpler is better.

Reply:
Can you suggest how? this is the simplest I could think of.
I thought of overloading the -> operator but too much overloading is not simple and I decided against it.
Do you mean by using some kind of overloaded operator?